src/Pages/Detail/Components/DetailHeader/HeaderPanelContainer/SelfRating/SelfRating.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react";
+import "./SelfRating.scss";
+
+const titleByRating = {
+  0: "평가하기",
+  0.5: "최악이에요",
+  1: "싫어요",
+  1.5: "재미없어요",
+  2: "별로에요",
+  2.5: "부족해요",
+  3: "보통이에요",
+  3.5: "볼만해요",
+  4: "재미있어요",
+  4.5: "훌륭해요!",
+  5: "최고에요!",
+};
+
+class SelfRating extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.state.widthByRating = `${props.rating * 20}%`;
+  }
+
+  initWidth = () => {
+    this.setState({ widthByRating: `${this.props.rating * 20}%` });
+  };
+
+  traceMouse = (() => {
+    let wait = false;
+    return ({ nativeEvent }, index) => {
+      if (!wait) {
+        if (nativeEvent.offsetX > 20)
+          this.setState({ widthByRating: `${index * 20}%` });
+        else this.setState({ widthByRating: `${(index - 0.5) * 20}%` });
+        wait = true;
+        setTimeout(() => {
+          wait = false;
+        }, 100);
+      }
+    };
+  })();
+
+  giveRating = ({ nativeEvent }, index) => {
+    let rating = nativeEvent.offsetX > 20 ? index : index - 0.5;
+    if (this.props.rating === rating) this.props.changeRating(0);
+    else this.props.changeRating(rating);
+  };
+
+  render() {
+    return (
+      <div className="SelfRating">
+        <div className="titleByRating">{titleByRating[this.props.rating]}</div>
+        <div className="starContainer" onMouseLeave={this.initWidth}>
+          <div className="unratedStars">
+            {[0, 0, 0, 0, 0].map((_, index) => (
```

Comment:
```suggestion
            {Array(5).fill().map((_, index) => (
```
0, 0, 0, 0, 0은 조금 어색해보여서 이 방법은 어떨까 싶습니다